src/babel/index.ts

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { NodePath, PluginObj } from '@babel/core';
+import * as BabelTypes from '@babel/types';
+const PACKAGE_NAME = 'react-loosely-lazy';
+const BUNDLER_CACHE_ID_KEY = 'getCacheId';
+const MODULE_ID_KEY = 'moduleId';
+const LAZY_METHODS = ['lazyForPaint', 'lazyAfterPaint', 'lazy'];
+
+const DEFAULT_OPTIONS: {
+  [key: string]: { ssr: boolean; defer: number };
+} = {
+  lazyForPaint: { ssr: true, defer: 0 },
+  lazyAfterPaint: { ssr: true, defer: 1 },
+  lazy: { ssr: false, defer: 2 },
+};
+
+export default function({
+  types: t,
+  template,
+}: {
+  types: typeof BabelTypes;
+  template: any;
+}): PluginObj {
+  return {
+    visitor: {
+      ImportDeclaration(
+        path: NodePath<BabelTypes.ImportDeclaration>,
+        state: { opts?: { client?: boolean } }
+      ) {
+        const { client } = state.opts || {};
+        const source = path.node.source.value;
+
+        if (source !== PACKAGE_NAME) {
+          return;
+        }
+
+        const bindingNames = LAZY_METHODS;
+        const bindings = bindingNames
+          .map(name => path.scope.getBinding(name))
+          .filter(binding => binding);
+
+        bindings.forEach((binding: any) => {
+          const lazyMethodName = binding.identifier.name;
+
+          binding.referencePaths.forEach((refPath: any) => {
```

Comment:
`referencePaths` are of type `NodePath[]`

Reply:
Moved all types concerns into separate issue #10